src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/ArtifactsSizeTests.cs

```diff
@@ -17,67 +17,54 @@
+    ExclusionsHelper exclusionsHelper = new ExclusionsHelper("ArtifactExclusions.txt", nameof(ArtifactsSizeTests));
 
-    public ArtifactsSizeTests(ITestOutputHelper outputHelper) : base(outputHelper)
-    {
-        if (File.Exists(BaselineFilePath))
-        {
-            string[] baselineFileContent = File.ReadAllLines(BaselineFilePath);
-            foreach (string entry in baselineFileContent)
-            {
-                string[] splitEntry = entry.Split(':', StringSplitOptions.TrimEntries);
-                Baseline[splitEntry[0]] = long.Parse(splitEntry[1]);
-            }
-        }
-        else
-        {
-            Assert.Fail($"Baseline file `{BaselineFilePath}' does not exist. Please create the baseline file then rerun the test.");
-        }
-    }
+    public ArtifactsSizeTests(ITestOutputHelper outputHelper) : base(outputHelper) {}
 
     [ConditionalFact(typeof(Config), nameof(Config.IncludeArtifactsSizeTests))]
-    public void CompareArtifactsToBaseline()
+    public void CheckZeroSizeArtifacts()
     {
-        Assert.False(string.IsNullOrWhiteSpace(Config.SourceBuiltArtifactsPath));
         Assert.False(string.IsNullOrWhiteSpace(Config.SdkTarballPath));
 
-        var tarEntries = ProcessSdkAndArtifactsTarballs();
-        ScanForDifferences(tarEntries);
-        UpdateBaselineFile();
+        ProcessTarball(Config.SdkTarballPath, SdkType);
 
-        // Must wait to report differences until after the baseline file is updated else a failure
-        // will cause the baseline file to not be updated.
+        exclusionsHelper.GenerateNewBaselineFile(updatedFileTag: null, NewExclusions);
+
+        // Wait to report differences until after the baseline file is updated. 
+        // Else a failure will cause the baseline file to not be updated.
         ReportDifferences();
     }
 
-    private Dictionary<string, long> ProcessSdkAndArtifactsTarballs()
+    private void ProcessTarball(string tarballPath, string type)
     {
         string tempTarballDir = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
         Directory.CreateDirectory(tempTarballDir);
 
-        Utilities.ExtractTarball(Config.SdkTarballPath!, tempTarballDir, OutputHelper);
-        Utilities.ExtractTarball(Config.SourceBuiltArtifactsPath!, tempTarballDir, OutputHelper);
+        Utilities.ExtractTarball(tarballPath, tempTarballDir, OutputHelper);
+
+        foreach (string filePath in Directory.EnumerateFiles(tempTarballDir, "*", SearchOption.AllDirectories))
+        {
+            string relativePath = filePath.Substring(tempTarballDir.Length + 1);
+            string processedPath = ProcessFilePath(relativePath);
 
-        Dictionary<string, long> tarEntries = Directory.EnumerateFiles(tempTarballDir, "*", SearchOption.AllDirectories)
-            .Where(filePath => !filePath.Contains("SourceBuildReferencePackages"))
-            .Select(filePath =>
+            if (new FileInfo(filePath).Length == 0)
```

Comment:
Consider consolidating the nested if check to reduce the amount of nesting and simplify the code.

```
            if (new FileInfo(filePath).Length == 0
                && !exclusionsHelper.IsFileExcluded(processedPath, type))
```